mono/mini/mini-x86.c

```diff
@@ -70,7 +70,42 @@ static mono_mutex_t mini_arch_mutex;
 #define CALLCONV_IS_STDCALL(sig) ((sig)->pinvoke && ((sig)->call_convention == MONO_CALL_STDCALL || (sig)->call_convention == MONO_CALL_THISCALL))
 #endif
 
-#define X86_IS_CALLEE_SAVED_REG(reg) (((reg) == X86_EBX) || ((reg) == X86_EDI) || ((reg) == X86_ESI))
+#define callee_saved_regs_count 3
+
+typedef struct _MonoX86CalleeSavedRegisters {
+	int count;
+	int bitset;
+	int push [callee_saved_regs_count];
+	int pop [callee_saved_regs_count];
+	int monolmf [callee_saved_regs_count + 1];          // +1 for EBP special case
+	int monolmf_offset [callee_saved_regs_count + 1];   // +1 for EBP special case
+} MonoX86CalleeSavedRegisters;
+
+const static MonoX86CalleeSavedRegisters
+callee_saved_regs = {
+	callee_saved_regs_count,                            // count
```

Comment:
What is the purpose of these changes ? Is the set of registers on x86 going to change in future ? Does it differ between OSes ? The new code looks harder to understand than the old one.

Reply:
Factoring out the repetition makes it clear what is different vs. the same in each reptition. i.e. just the registers.